dart/lib/src/sentry_client.dart

```diff
@@ -20,22 +31,19 @@ class SentryClient {
     if (options.transport is NoOpTransport) {
       options.transport = HttpTransport(options);
     }
+
     return SentryClient._(options);
   }
 
-  final SentryOptions _options;
-
-  final Random _random;
-
-  final SentryExceptionFactory _exceptionFactory;
-
-  static final _sentryId = Future.value(SentryId.empty());
-
   /// Instantiates a client using [SentryOptions]
-  SentryClient._(this._options, {SentryExceptionFactory exceptionFactory})
-      : _exceptionFactory =
-            exceptionFactory ?? SentryExceptionFactory(options: _options),
-        _random = _options.sampleRate == null ? null : Random();
+  SentryClient._(this._options)
```

Comment:
they (SentryStackTraceFactory, SentryExceptionFactory) should still be optional param for testing I guess? but only assign a value if they are null

Reply:
sorry but I'm not sure to understand, should we add factories param to SentryClient factory ctor and private ctor to allow users to pass their own factories ?

Reply:
no, the reasoning is not to allow custom factories, the reason is to make it easily testable, if you can unit test it like this (without fakes or mocks), its fine.

Reply:
seems ok (for current tests)